### PR TITLE
Paid master nodes

### PR DESCRIPTION
**Pull request by oskarth**
Rough start, but enough to get some discussion going.

Can also talk in #168-paid-master-nodes 

- [x] should be 168

### CODE REVIEW

File: ideas/167-paid-master-node.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+## Product Overview
+
+This idea takes of where https://github.com/status-im/ideas/blob/master/ideas/1-offline-inboxing.md ended. It provides an SNT payment layer for usage of master nofdes.
+
+> We want to make it easy to run a Whisper mail server (part of Whisper v5 API). This will archive messages and forward them to a peer upon request. You can run this yourself if you want to, but we’ll also provide default mail servers through our clusters, and make it easy for anyone to set one up. This can be paid using SNT, which aligns incentives in terms of Quality of Service.
+
+We need something users can do with their SNT, to realize the power of the
+technology and keep them interested.
+
+Currently offline inboxing is a free resource, subsidies and run by us. This is
```
Comment (andytudhope) [inline ideas/167-paid-master-node.md]: `subsidised`

File: ideas/167-paid-master-node.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+Offline inboxing is run on mail server aka master nodes. In the scope of e.g.
+86-push-notifications push notifications will likely be served through that as
+well.
+
+### Constraints / objections:
+
+- No one will pay for it.
+
+This is a free resource. If we want to subsidies it and the concern is people
```
Comment (andytudhope) [inline ideas/167-paid-master-node.md]: `subsidised`

File: ideas/167-paid-master-node.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+Solving the two initial user stories. If the swarm deems it useful to continue,
+new specific user stories will be specified and inform the exit criteria.
+
+## Supporting Role Communication
+
+- Community for recruiting users to run mail server.
+
+- Finance for SNT utility approval.
+
+- Marketing for 'you are not the product' rationale communication.
```
Comment (andytudhope) [inline ideas/167-paid-master-node.md]: Happy to handle this part 👍

File: ideas/167-paid-master-node.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+- QA: TBD
+- Evaluation: TBD
+- Clojure Contributor: TBD
+- Go Contributor: TBD
+- PM: TBD
+- UX(R): TBD
+
+## Product Overview
+
+This idea takes of where https://github.com/status-im/ideas/blob/master/ideas/1-offline-inboxing.md ended. It provides an SNT payment layer for usage of master nofdes.
```
Comment (dmitryn) [inline ideas/167-paid-master-node.md]: `nofdes` type

**Comment by JekaMas** [comment]
This should be implemented in 95 swarm about les service model. AFAIK
@themue

**Comment by themue** [comment]
Yeah, it's overlapping with https://github.com/status-im/ideas/tree/master/ideas/095-les-service-model and could benefit of it. While #95 is focussed on providing light clients a higher service quality by paying according full nodes it so far doesn't cover additional services like mailboxing. But IMHO that should be relative easy to integrate. Will get Zsolt into discussion to keep API open enough for future services.

Thanks for pinging, @JekaMas .

File: ideas/167-paid-master-node.md (hunk 3 above)
Comment (oskarth) [reply ideas/167-paid-master-node.md]: awesome! added

File: ideas/167-paid-master-node.md (hunk 2 above)
Comment (oskarth) [reply ideas/167-paid-master-node.md]: fixed

File: ideas/167-paid-master-node.md (hunk 1 above)
Comment (oskarth) [reply ideas/167-paid-master-node.md]: fixed

File: ideas/167-paid-master-node.md (hunk 4 above)
Comment (oskarth) [reply ideas/167-paid-master-node.md]: fixed

**Comment by oskarth** [comment]
@dmitryn Oops, fixed!

@JekaMas @themue They are sufficiently different in focus, approach and end product to warrant two approaches imo (research-oriented, ULC dependency, more low-level / general, possibly longer timeline, etc).

There's also nothing wrong with competing ideas :)

File: ideas/167-paid-master-node.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+SNT also furthers distribution of tokens as well as transactions on their own, a
+good.
+
+This is a hypothesis, but this is the whole basis of Status.
+
+### User stories
+
+User story 1: Bobby Blockchain runs a master node and gets paid by peers in SNT for services provided.
+
+User story 2: Alice pays SNT to a master node in order to get offline inboxing for some period of time.
```
Comment (jpbowen) [inline ideas/167-paid-master-node.md]: May be out of scope for this idea but perhaps we'll want to experiment with different payment models for time - e.g. Alice pays until she runs out of SNT OR N offline messages OR set period of time.

File: ideas/167-paid-master-node.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+such as communication of quality of service and price discovery mechanisms. It
+is unlikely these will be relevant until user story 1 and 2 are solved, but this
+swarm will decide if it makes sense to continue working to solve additional user
+stories after 1 and 2 are done.
+
+### Product Description
+<!-- What functionality are you adding? What will this look like from a user perspective? Why is this important? -->
+
+Currently we have a screen that shows mail servers available. Each mail server
+will have a basic monthly fee attached to it. This can start off as static but
```
[the remark this answers is not in the document]
Comment (jpbowen) [reply ideas/167-paid-master-node.md]: +1

File: ideas/167-paid-master-node.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+For more on rationale of SNT utility, see the whitepaper. Providing onboarding
+SNT also furthers distribution of tokens as well as transactions on their own, a
+good.
+
+This is a hypothesis, but this is the whole basis of Status.
+
+### User stories
+
+User story 1: Bobby Blockchain runs a master node and gets paid by peers in SNT for services provided.
```
Comment (rachelhamlin) [inline ideas/167-paid-master-node.md]: What are the minimum requirements to run a wnode? Is there an initial stake?

File: ideas/167-paid-master-node.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+Out of scope for this idea:
+
+- (User story 3: Decentralized discovery mechanism for master nodes)
+
+- (User story 4: Onboarding mechanism to seed closed beta users with SNT)
+
+User story 4 is sufficiently different in focus to warrant a separate swarm.
+Since it might be a pre-requsite from a UX point of view, the behavior in user
+story 1 and 2 should be toggle-able with a flag. I.e. we make sure we can still
+provide free service for the time being.
```
Comment (rachelhamlin) [inline ideas/167-paid-master-node.md]: How will we encourage users to try this if it remains available for free? Agreed a soft launch is better but perhaps requires a beta user campaign.

File: ideas/167-paid-master-node.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+users.
+
+## Swarm Participants
+
+- Lead Contributor: @oskarth
+- QA: TBD
+- Evaluation: TBD
+- Clojure Contributor: TBD
+- Go Contributor: TBD
+- PM: TBD
```
[the remark this answers is not in the document]
Comment (rachelhamlin) [reply ideas/167-paid-master-node.md]: Sure, sign me up 🙋‍♀️

File: ideas/167-paid-master-node.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+- UX(R): TBD
+- Community: @andytudhope
+
+## Product Overview
+
+This idea takes of where https://github.com/status-im/ideas/blob/master/ideas/1-offline-inboxing.md ended. It provides an SNT payment layer for usage of master nodes.
+
+> We want to make it easy to run a Whisper mail server (part of Whisper v5 API). This will archive messages and forward them to a peer upon request. You can run this yourself if you want to, but we’ll also provide default mail servers through our clusters, and make it easy for anyone to set one up. This can be paid using SNT, which aligns incentives in terms of Quality of Service.
+
+We need something users can do with their SNT, to realize the power of the
```
[the remark this answers is not in the document]
Comment (oskarth) [reply ideas/167-paid-master-node.md]: replaced with user-center paragraph

File: ideas/167-paid-master-node.md [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+Currently offline inboxing is a free resource, subsidised and run by us. This is
+sub-optimal for many reasons, and goes against our vision of letting users pay
+for things they use, as well as incentivizing this capability to be
+decentralized.
+
+Offline inboxing is run on mail server aka master nodes. In the scope of e.g.
+86-push-notifications push notifications will likely be served through that as
+well.
+
+### Constraints / objections:
```
[the remark this answers is not in the document]
Comment (oskarth) [reply ideas/167-paid-master-node.md]: moved to appendix

File: ideas/167-paid-master-node.md [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+User story 1: Bobby Blockchain runs a master node and gets paid by peers in SNT for services provided.
+
+User story 2: Alice pays SNT to a master node in order to get offline inboxing for some period of time.
+
+Out of scope for this idea:
+
+- (User story 3: Decentralized discovery mechanism for master nodes)
+
+- (User story 4: Onboarding mechanism to seed closed beta users with SNT)
```
[the remark this answers is not in the document]
Comment (oskarth) [reply ideas/167-paid-master-node.md]: Just want to outline them here in parentheses to show how they are related. It is bound to come up, so it is good to make it explicit that it is out of scope for now.

Re early adopter vs onboarding point in question: fair point! let's discuss outside this idea

File: ideas/167-paid-master-node.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (oskarth) [reply ideas/167-paid-master-node.md]: Great point, we definitely want this but perhaps not in iteration 1. This is a great thing we can find out with UXR and include in further iterations imo.

Adding as a note in idea

File: ideas/167-paid-master-node.md [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+technical terms on status-go side.
+
+#### Relevant reading
+
+- https://status.im/whitepaper.pdf
+- http://nakamotoinstitute.org/static/docs/micropayments-and-mental-transaction-costs.pdf
+
+### Requirements & Dependencies
+
+Possibly onboarding SNT give-away logistics.
```
[the remark this answers is not in the document]
Comment (oskarth) [reply ideas/167-paid-master-node.md]: edited to reflect lack of consensus, broader discussion (doesn't stop work in this swarm, just in terms of when we flip the switch)

File: ideas/167-paid-master-node.md [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+
+### Minimum Viable Product
+<!-- Mandatory, completes the Idea in the fastest route possible, can be hacky, needed to feel progress. See https://imgur.com/a/HVlw3 -->
+Goal Date: TBD - May 1 + 2w
+
+Description: Spike out payment Clojure and Go PoC with STT.
```
[the remark this answers is not in the document]
Comment (oskarth) [reply ideas/167-paid-master-node.md]: replaced by longer sentence, essentially just hacking together most basic proof of concept in code only

**Comment by oskarth** [comment]
Addressed feedback and added review notes in idea, merging as draft.

Let's continue in #168-paid-master-nodes :)

File: ideas/168-paid-master-nodes.md [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+  
+- Free SNT?!
+
+Yes. For a closed beta limited to 40k people, we can limit the free amount to 5
+SNT per person, which limits the downside risk this to 200k SNT ($20k at the
+time of this writing).
+
+- What about fraud?
+
+We can use already collected emails as identity and dedup by it. Max downside
```
Comment (PombeirP) [inline ideas/168-paid-master-nodes.md]: We'll probably want to blacklist some proxy email domains such as maskr.me, 33mail.com, etc. which allow you to create unlimited email accounts.

**Pull request merged**